PublishParasoftResults/ParaReportPublishService.ts

```diff
@@ -576,18 +576,27 @@ export class ParaReportPublishService {
 
     private async processCoberturaResults(): Promise<void> {
         if (this.coberturaReports.length > 0) {
-            const tempFolder = path.join(this.getTempFolder(), 'CodeCoverageHtml');
+            const parasoftFindingsTempFolder = path.join(this.getTempFolder(), 'ParasoftFindings')
             const coverageReportService = new CoverageReportService();
-            const coverageReport = coverageReportService.mergeCoberturaReports(this.coberturaReports);
-            if (!coverageReport) {
-                tl.warning('No merged coverage report generated.');
-                return;
+
+            // Check and merge Cobertura report from artifacts
+            const mergedCoberturaReportFromArtifacts = await coverageReportService.getMergedCoberturaReportByBuildId(Number(this.buildId));
+            if (mergedCoberturaReportFromArtifacts) {
+                const mergedCoberturaReportFileFromArtifacts =  parasoftFindingsTempFolder + "/parasoft-merged-cobertura-from-artifact.xml";
+                fs.writeFileSync(mergedCoberturaReportFileFromArtifacts, await mergedCoberturaReportFromArtifacts.contentsPromise, 'utf-8');
+                this.coberturaReports.unshift(mergedCoberturaReportFileFromArtifacts);
```

Comment:
`unshift` can be inefficient for large arrays because it requires shifting all the existing elements. 
Can we improve this? For example, passing the base report as an argument to the merge method could be a solution.